### ISSUE TITLE
Getting the Initial focus in the right place

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4686]
Some of the dialogues have the initial focus in an odd place.  And it would save a click, but the dialogue would also be more obvious to use if this were improved.

This is more important here than for other software, because we are aiming for beginners.  I suggest we go through this, partly during our testing on Version 4.18 and also in Malawi.  I also assume it is easy to fix:
1. As an example look at Climatic > Check Data > Inventory Plot

There are just 3 fields.  2 are automatically filled, namely Date and Station.  The initial focus is on Station.  The only field to be filled is Elements so to initial focus should be there.

2. Now the Display Daily Data

The initial focus is on Station.  I had Dodoma data, where there was no station variable.  The only variable to complete was Element, so this should have the focus.  I click on Element and it is Turquise.  Nice.  Then I click on the Graph button at the top to change to a graph and nothing is now highlighted.  Perhaps it isn't so simple?

Can we "fix" these and look steadily for others?

**Comment by dannyparsons** [comment]
I think this should be easy and for almost all climatic dialogs the element can be the initial focus

**Comment by shadrackkibet** [comment]
Working on this.

**Comment by Muthenya** [comment]
I've taken this up from @shadrackkibet and I've made some very interesting finds.

The order in which you set up receivers (I guess this applies to all controls) on the designer and the **bAutoFill** property of the receiver influence its focus.
The way around the issue (maybe for now) is;
1. Ensuring that, whichever the position of the **Element Receiver**, it be the first to be set up on the designer.
2. The **bAutoFill** property for the **Element Receiver** be set to **True** even if it won't be automatically filled.

3. Now for the case of **multiple receivers**, since they do not have the **bAutoFill** property,  one way around this is to not set the **bAutoFill** property to **True** for other (single) receivers.

@africanmathsinitiative/developers please let me get your take on this.

**Comment by dannyparsons** [comment]
At what point is the auto filling happening? Isn't it when the data frame changes? If so then setting the current receiver can be done after autofilling has happened.

**Comment by Muthenya** [comment]
@dannyparsons I tried this but it doesn't give what we want and it still doesn't answer the issue of **Multiple Receiver**.

**Comment by dannyparsons** [comment]
Did you find out at what point in the loading of the dialog the auto filling is happening? That will help to be able to control this.

**Comment by Muthenya** [comment]
This happens during the loading of a single receiver.

**Comment by dannyparsons** [comment]
Because the auto filling happens are different times, including when the form is first shown, it's not easy to set the current receiver after this has happened. So I have modified the auto filling code so that it doesn't modify the current receiver. This means that setting the current receiver in SetDefaults() won't change by the auto filling. I tested this on Display Daily and works fine but check other dialogs. I didn't check dialogs where we are putting in a default, like rain for element, but since this is dialog code it's easy to modify to get the current receiver to not be changed if needed.
